### PR TITLE
Datasource analysis in separate thread

### PR DESCRIPTION
**Pull request by StpMax**
Reason: when user push on 'data quality' in gui, it call `/ds_name/analyze` route. This query will be active untill analysis finished. It can take much time for big datasets, query can fail by timeout. As resolve i done this:
1. when first time called `/ds_name/analyze`, analysis starts and returned to user {status: 'analyzing'}
2. {status: 'analyzing'} will be returned each time while analysis not finished
3. then if finished, returned analysis data

### CODE REVIEW

**Comment by ZoranPandovski** [comment]
**CLA Assistant Lite:** <br/>Thank you for your submission, we really appreciate it. Like many open-source projects, we ask that you sign our [Contributor License Agreement](https://github.com/mindsdb/mindsdb_native/blob/stable/assets/contributions-agreement/individual-contributor.md) before we can accept your contribution. You can sign the CLA by just posting a Pull Request Comment same as the below format.
  - - -
  ***I have read the CLA Document and I hereby sign the CLA***
  - - - 
  <sub>You can retrigger the CLA Action by commenting **recheckcla** in this Pull Request</sub>

**Comment by StpMax** [comment]
I have read the CLA Document and I hereby sign the CLA

File: mindsdb/api/http/namespaces/datasource.py [hunk 1]
```diff
@@ -117,19 +118,43 @@ def on_file(file):
...
             abort(400, 'No valid datasource given')
 
-        analysis = ca.default_store.get_analysis(ds['name'])
-
-        return analysis, 200
+        if ds['row_count'] <= 10000:
+            analysis = ca.default_store.get_analysis(ds['name'])
+            return analysis, 200
+        else:
+            x = threading.Thread(target=analyzing_thread, args=(name, ca.default_store))
```
Comment (George3d6) [inline mindsdb/api/http/namespaces/datasource.py:155]: Does using a thread here actually help ? I'd think it's better to just use a subprocess instead in a similar way we do it for `learn` and `predict`, for now feel free to merge it like this, but if you have time please create an issue around migrating this to a subprocess based approach (via python's multiprocessing, that is).

Comment (George3d6) [reply mindsdb/api/http/namespaces/datasource.py:155]: Also, there is currently a "mis usage" of the data analysis in that we refresh the analysis cache to often via `del ds_analysis[name]`. We should consider removing this or adding it behind a time (e.g. only refresh if the analysis is older than 2 days), could you add this functionality as part of this PR ?

**Pull request merged**